lib/tokenize.cpp

```diff
@@ -10792,6 +10795,71 @@ void Tokenizer::simplifyDeclspec()
     }
 }
 
+void Tokenizer::simplifyAttributeList()
+{
+    for (Token *tok = list.front(); tok; tok = tok->next()) {
+        while (Token::Match(tok, "__attribute__|__attribute (") && 
+				tok->next()->link() != tok->next()->next() &&
+				tok->next()->link() && tok->next()->link()->next()) {
```

Comment:
I don't think it's necessary to check `tok->next()->link()`.. it is checked that there is a matching ")" token.